### PR TITLE
Basic support for slackclient v2

### PR DESCRIPTION
**Pull request by gbin**
It adds a LOT of noise in Errbot as the API is way lower level than the previous version (this is basically a pure socket to Slack + json parsing).

### CODE REVIEW

File: errbot/backends/slack.py [hunk 1]
```diff
@@ -135,14 +147,19 @@ def channelname(self):
...
             raise RoomDoesNotExistError(f'No channel with ID {self._channelid} exists.')
-        return channel.name
+        if not self._channelname:
+            self._channelname = channel['name']
+        return self._channelname
 
     @property
     def domain(self):
-        return self._sc.server.domain
+        raise NotImplemented()
```
Comment (dizzi90) [inline errbot/backends/slack.py]: Can this be read from [scopes/identity.team:read:user](https://api.slack.com/scopes/identity.team:read:user)?

**Comment by sijis** [comment]
I'm wondering if this should be a new backend as `slack-rtm` (or something).

This way we can cut a release with a 'beta' tag on this backend. With user usage we can improve on it and later deprecate the `slack` backend once its more mature.

Thoughts?

**Comment by sijis** [comment]
I forked the changes from here into https://github.com/errbotio/errbot/pull/1367 and registered it as a new backend `SlackRTM`.

**Comment by flokli** [comment]
@sijis where was this merged into? #1367 is this issue here.

**Comment by sijis** [comment]
@flokli sorry about that. I updated the link. It's not merged yet.

**Comment by sijis** [comment]
#1416 was merged and released in 6.1.3.

**Pull request closed without merging**